src/api/procedures/modifyMultiSig.ts

```diff
@@ -0,0 +1,101 @@
+import { MultiSig } from '~/api/entities/MultiSig';
+import { PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, Signer, TxTags } from '~/types';
+import { signerToSignatory, signerToString, stringToAccountId } from '~/utils/conversion';
+
+/**
+ * @hidden
+ */
+export interface ModifyMultiSigParams {
+  multiSig: MultiSig;
+  signers: Signer[];
+}
+
+/**
+ * @hidden
+ */
+function calculateSignerDelta(current: Signer[], target: Signer[]): [Signer[], Signer[]] {
+  const currentSet = new Set(current.map(signerToString));
+  const targetSet = new Set(target.map(signerToString));
+
+  const newSigners = new Set([...target].filter(s => !currentSet.has(signerToString(s))));
+  const removedSigners = new Set([...current].filter(s => !targetSet.has(signerToString(s))));
+
+  return [Array.from(newSigners), Array.from(removedSigners)];
+}
+
+/**
+ * @hidden
+ */
+export async function prepareModifyMultiSig(
+  this: Procedure<ModifyMultiSigParams, void>,
+  args: ModifyMultiSigParams
+): Promise<void> {
+  // adds or removes signers and can change number of signatures required.
+  // Should also provide a "as owner flag" to allow for a user to "override" other signatures
+  const {
+    context: {
+      polymeshApi: { tx },
+    },
+    context,
+  } = this;
+  const { signers, multiSig: multi } = args;
+
+  // ensure calling account is creator, other actions require a proposal from the multisig
+  const signingIdentity = await context.getSigningIdentity();
+  const creator = await multi.getCreator();
+  if (!creator.isEqual(signingIdentity)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'A MultiSig can only be modified by its creator',
```

Comment:
Do you mean put this check in the branch this is based off of?

Right now we can't perform any action to change the multiSig unless this is the case though. This check makes sense to me.

Reply:
In the future, we will support calling this procedure as a proposal instead of just as the creator. When we do, we will need to remove this error. I'm worried we might forget to do so

Reply:
I am using the transactions `tx.multiSig.addMultisigSignersViaCreator` and `tx.multiSig.removeMultisigSignersViaCreator`.

We might need to expose the non ViaCreator variations, but either that should get its own method, or we will need to modify this one, in which this error will be found.

Reply:
To add on. If making a proposal, then the proposal should be for `multiSig.addMultiSigSigner`, and not for this method.

Either we make a new method for the proposal version, or we make this one use a dynamic extrinsic depending on the circumstance.